connectors/sources/google_drive.py

```diff
@@ -589,29 +672,85 @@ def _validate_google_workspace_admin_email(self):
+                - If the Google Workspace email for shared drives sync is empty when the domain-wide delegation sync is enabled.
+                - If the provided email address is malformed or contains whitespace characters.
+        """
+        if self._domain_wide_delegation_sync_enabled():
+            google_workspace_email = self.configuration[
+                "google_workspace_email_for_shared_drives_sync"
+            ]
+
+            if google_workspace_email is None:
+                raise ConfigurableFieldValueError(
+                    "Google Workspace admin email for shared drives sync cannot be empty when 'Use domain-wide delegation for data sync' is enabled."
+                )
+
+            if not validate_email_address(google_workspace_email):
+                raise ConfigurableFieldValueError(
+                    "Google Workspace email for shared drives sync is malformed or contains whitespace characters."
+                )
+
     async def ping(self):
         """Verify the connection with Google Drive"""
         try:
-            await self.google_drive_client.ping()
+            if self._domain_wide_delegation_sync_enabled():
+                admin_email = self._get_google_workspace_admin_email()
+                await self.google_drive_client(impersonate_email=admin_email).ping()
+            else:
+                await self.google_drive_client().ping()
             self._logger.info("Successfully connected to the Google Drive.")
         except Exception:
             self._logger.exception("Error while connecting to the Google Drive.")
             raise
 
+    def _get_google_workspace_admin_email(self):
+        """
+        Retrieves the Google Workspace admin email based on the current configuration.
+
+        If domain-wide delegation for data sync is enabled, this method will return the admin email
+        provided for shared drives sync. If Document Level Security (DLS) is enabled but not domain-wide
+        delegation, it will return the the admin email specified for DLS.
+
+        This ensures that if the admin email for domain-wide delegation is provided, it is utilized
+        for both sync and DLS without requiring the same email to be provided again for DLS.
+
+        Returns:
+            str or None: The Google Workspace admin email based on the current configuration or None if
+            neither domain-wide delegation nor DLS is enabled.
+        """
+
+        if self._domain_wide_delegation_sync_enabled():
+            return self.configuration["google_workspace_admin_email_for_data_sync"]
+        elif self._dls_enabled():
+            return self.configuration["google_workspace_admin_email"]
+        else:
+            return None
+
+    def _google_google_workspace_email_for_shared_drives_sync(self):
+        """Get Google Workspace email for syncing shared drives"""
+        return self.configuration.get("google_workspace_email_for_shared_drives_sync")
```

Comment:
I don't know, if the comment adds any value here and it should probably be `_get_google[...]` and not `_google_google`?

```suggestion
    def _get_google_workspace_email_for_shared_drives_sync(self):
        return self.configuration.get("google_workspace_email_for_shared_drives_sync")
```